### PR TITLE
Add more parameters for connection add

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added new automated tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [x] I've described my changes in the section below.

### Changes description
Add more parameters to `snow connection add` to support all basic options for connection configuration.

### CODE REVIEW

File: src/snowcli/cli/connection.py [hunk 1]
```diff
@@ -36,38 +48,89 @@ def add(
...
         "account": account,
         "user": user,
         "password": password,
+        "host": host,
+        "region": region,
+        "port": port,
+        "database": database,
+        "schema": schema,
     }
+    print(connection_entry)
```
Comment (sfc-gh-astus) [inline src/snowcli/cli/connection.py:131]: debug print? 😄

File: src/snowcli/cli/connection.py [hunk 2]
```diff
@@ -36,38 +48,89 @@ def add(
...
     ),
     account: str = typer.Option(
         None,
         "-a",
         "--accountname",
         "--account",
-        prompt="Snowflake account",
-        help="Name assigned to your Snowflake account.",
+        help="Account name to be used to authenticate with Snowflake.",
+        show_default=False,
```
Comment (sfc-gh-astus) [inline src/snowcli/cli/connection.py:59]: What happened with account? Why removed prompt?

File: src/snowcli/cli/connection.py [hunk 3]
```diff
@@ -36,38 +48,89 @@ def add(
...
+        help="The port to communicate with on the host.",
+    ),
+    region: str = typer.Option(
+        EmptyInput(),
+        "-r",
+        "--region",
+        click_type=OptionalPrompt(),
+        prompt="Snowflake region",
+        help="Region name if not the default Snowflake Database deployment.",
+    ),
```
Comment (sfc-gh-astus) [inline src/snowcli/cli/connection.py:118]: First time see region, do we need this one? 

What about role, warehouse and protocol?

Comment (sfc-gh-turbaszek) [reply src/snowcli/cli/connection.py:118]: Yup, region is supported by snowflake python connector. Good point about role and wh which are more important. Added!

File: src/snowcli/cli/connection.py (hunk 2 above)
Comment (sfc-gh-turbaszek) [reply src/snowcli/cli/connection.py:60]: Good catch, fixed

File: src/snowcli/cli/connection.py (hunk 3 above)
Comment (sfc-gh-astus) [reply src/snowcli/cli/connection.py:135]: My eyes bleeding looking at parameters and connection_entry order but iiyo

**Review by sfc-gh-astus — APPROVED** [review]

**Pull request merged**